src/main/java/com/google/graphgeckos/dashboard/fetchers/buildbot/BuildBotClient.java

```diff
@@ -0,0 +1,78 @@
+package com.google.graphgeckos.dashboard.fetchers.buildbot;
+
+import com.google.graphgeckos.dashboard.datatypes.BuildBotData;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.graphgeckos.dashboard.storage.DatastoreRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.lang.NonNull;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+/** A (external) BuildBot API json data fetcher. */
+public class BuildBotClient {
+
+  /** Provides access to the storage. */
+  @Autowired
+  private DatastoreRepository datastoreRepository;
+
+  /** Base url of the BuildBot API. */
```

Comment:
Consider adding a more in-depth explanation of what the base URL is here. Even though it is explained in the constructor, I believe this should be clear in the field explanation as well.

Reply:
Base url is self-explanatory.

Reply:
I think it's worth repeating the base URL part of the example from the constructor here. Specifically, consider this extract from the example:
>baseUrl = "http://lab.llvm.org:8011/json/builders"
Someone not familiar with the LLVM BuildBot URLs can reasonably think they can pass any of the following as baseUrl:
* http://lab.llvm.org:8011
* http://lab.llvm.org:8011/json
* http://lab.llvm.org:8011/json/builders